pc/user.cpp

```diff
@@ -330,17 +330,27 @@ void user::parse_get_product( uint32_t tok, uint32_t itok )
   add_tail( itok );
 }
 
-void user::send_pending_upds()
+uint32_t user::num_pending_upds()
+{
+  return pending_vec_.size();
+}
+
+void user::send_pending_upds(uint32_t n)
```

Comment:
Regarding testing: we don't have an easy way to reliably test _when_ transactions are sent. To test _if_ partial batches and full batches are sent, you could use something like [test_update_price.py](https://github.com/pyth-network/pyth-client/blob/main/pyth/tests/test_update_price.py).